webapp/channels/src/components/admin_console/team_channel_settings/group/group_list.tsx

```diff
@@ -12,57 +12,62 @@ import GroupRow from './group_row';
+        <div className='group-name group-name-adjusted'>
+            <FormattedMessage
+                id='admin.team_channel_settings.group_list.nameHeader'
+                defaultMessage='Group Name'
+            />
+        </div>
+        <div className='group-content'>
+            <div className='group-description group-description-adjusted'>
                 <FormattedMessage
-                    id='admin.team_channel_settings.group_list.nameHeader'
-                    defaultMessage='Group Name'
+                    id='admin.team_channel_settings.group_list.membersHeader'
+                    defaultMessage='Member Count'
                 />
             </div>
-            <div className='group-content'>
-                <div className='group-description group-description-adjusted'>
-                    <FormattedMessage
-                        id='admin.team_channel_settings.group_list.membersHeader'
-                        defaultMessage='Member Count'
-                    />
-                </div>
-                <div className='group-description group-description-adjusted'>
-                    <FormattedMessage
-                        id='admin.team_channel_settings.group_list.rolesHeader'
-                        defaultMessage='Roles'
-                    />
-                </div>
-                <div className='group-actions'/>
+            <div className='group-description group-description-adjusted'>
+                <FormattedMessage
+                    id='admin.team_channel_settings.group_list.rolesHeader'
+                    defaultMessage='Roles'
+                />
             </div>
+            <div className='group-actions'/>
         </div>
-    );
-};
+    </div>
+);
 
 type Props = OwnProps & PropsFromRedux;
 
-export default class GroupList extends React.PureComponent<Props> {
-    renderRow = (item: Partial<Group>) => {
+const GroupList = (props: Props) => {
+    const {
+        removeGroup,
+        setNewGroupRole,
+        type,
+        isDisabled,
+    } = props;
+
+    const renderRow = (item: Partial<Group>) => {
         return (
             <GroupRow
                 key={item.id}
                 group={item}
-                removeGroup={this.props.removeGroup}
-                setNewGroupRole={this.props.setNewGroupRole}
-                type={this.props.type}
-                isDisabled={this.props.isDisabled}
+                removeGroup={removeGroup}
+                setNewGroupRole={setNewGroupRole}
+                type={type}
+                isDisabled={isDisabled}
             />
         );
     };
 
-    render(): JSX.Element {
-        return (
-            <AbstractList
-                header={<Header/>}
-                renderRow={this.renderRow}
-                {...this.props}
-            />
-        );
-    }
-}
+    return (
+        <AbstractList
+            header={<Header/>}
+            renderRow={renderRow}
+            {...props}
```

Comment:
I feel we are passing here many things to abstract list that are not needed.
In particular, none of the props we are using here (`removeGroup`, `setNewGroupRole`, `type` or `isDisabled`) are expected in AbstractList. So we can change how we expand the props to do it like...

```
const GroupList = ({
    removeGroup,
    ...,
    ...restProps
}: Props) => {
    ...
    return (<AbstractList header={<Header/>} renderRow={renderRow} {...restProps} />);
};
```